src/main/java/macrobase/transform/FeatureTransform.java

```diff
@@ -0,0 +1,39 @@
+package macrobase.transform;
+
+import macrobase.conf.MacroBaseConf;
+import macrobase.datamodel.Datum;
+import macrobase.ingest.DataIngester;
+
+import javax.xml.crypto.Data;
+import java.util.Iterator;
+import java.util.function.Consumer;
+
+public class FeatureTransform implements Iterator<Datum> {
+
+    DataIngester ingester;
+
+    public FeatureTransform(MacroBaseConf conf, DataIngester ingester) {
+        // TODO: apply specified configurations
+        this.ingester = ingester;
+    }
+
+    /**
+      * Applies the transformation this class represents to the datum given and
+      * outputs a new transformed datum.
+      * @param datumToTransform datum that should be transformed
+      * @return new Datum that has transformation applied to it.
+      */
+    public Datum transform(Datum datumToTransform) {
```

Comment:
What's the rationale? 

Say I want to link to MacroBase and implement my own `FeatureTransform`. Do I need to place my `FeatureTransform` in the `macrobase.transform` package?

(It's possible I am misunderstanding; it's late :v:.)